api/server.go

```diff
@@ -54,6 +55,12 @@ func (s *Server) WithAuth(authProvider func(*http.Request) (string, error)) {
 	}
 }
 
+// WithCustomMiddlewares sets an array of customized gin middlewares.
+// It helps for init plugins to include these customized middlewares in the api server
+func (s *Server) WithCustomMiddlewares(customMiddlewares ...gin.HandlerFunc) {
+	s.customMiddlewares = customMiddlewares
```

Comment:
I wonder if multiple plugins starts to add middlewares, they will break previous plugins middlewares.

```suggestion
	s.customMiddlewares = append(s.customerMiddlewares, customMiddlewares...)
```
This suggestion might fix it. What do you think @zhouzhuojie @loopfz ?